src/consensus/tx_verify.cpp

```diff
@@ -289,6 +291,27 @@ bool CheckTransaction(const CTransaction& tx, CValidationState &state, int nHeig
         nValueOut += txout.nValue;
         if (!MoneyRange(nValueOut, isV17active))
             return state.DoS(100, false, REJECT_INVALID, "bad-txns-txouttotal-toolarge");
+
```

Comment:
can we move these assets related check into its own function so it a bit easier to follow? this function is getting longer as time goes one.